command/operator_api.go

```diff
@@ -0,0 +1,446 @@
+package command
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/nomad/api"
+	"github.com/posener/complete"
+)
+
+type OperatorAPICommand struct {
+	Meta
+
+	verboseFlag bool
+	method      string
+	body        io.Reader
+}
+
+func (*OperatorAPICommand) Help() string {
+	helpText := `
+Usage: nomad operator api [options] <path>
+
+  api is a utility command for accessing Nomad's HTTP API and is inspired by
+  the popular curl command line program. Nomad's operator api command populates
+  Nomad's standard environment variables into their appropriate HTTP headers.
+  If the 'path' does not begin with "http" then $NOMAD_ADDR will be used. 
+
+  The 'path' can be in one of the following forms:
+
+    /v1/allocations                       <- API Paths must start with a /
```

Comment:
would it make sense to require paths start with `/v1/`?

e.g. hitting the UI is a little funny 
```
➜ nomad operator api /
<!DOCTYPE html>
<html>
```

Reply:
Not a bad idea, but my bias is toward leaving it as-is. I'd like to try to constrain this tool to `curl but with Nomad env vars and parameters`. In that vein the principle behind allowing `/` is that we'll inject the host if `-address` or `$NOMAD_ADDR` is set, but otherwise we're just curl. Maybe they hit the UI in a browser and got an error due to mTLS so they want to test it via the CLI?

Obviously my take is very subjective, so please let me know if it doesn't make sense. Definitely want this tool to be as useful as possible!

Reply:
:+1: